### PR TITLE
feat: enable Google Polyline in Static Images endpoint 🕺

### PR DESCRIPTION
**Pull request by vinayakkulkarni**
- [x] Based off the work from Niklas in #209, this PR enables `encodedpath` query parameter which accepts Google encoded polyline, decodes its paths & renders the static tile with correct polyline.
- [x] Updated the `path` query param to match [Maptiler API](https://docs.maptiler.com/cloud/api/static-maps/)
- [x] Updated docs

#### Google Encoding
- [x] [fill, width, stroke, enc](http://localhost:8080/styles/osm-bright/static/auto/400x400.png?path=fill:%233f34|width:4|stroke:%233339|enc:ovvkC}ezyLEaC|DpRzDvRhBhLld@fO|]fUna@lVjb@nPnSgPnOqVlSNNeA|LKtGa@jXgF?CjDYx^qG|JoJ) 
- [x] [width, stroke, enc](http://localhost:8080/styles/osm-bright/static/auto/400x400.png?path=width:4|stroke:%233339|enc:ovvkC}ezyLEaC|DpRzDvRhBhLld@fO|]fUna@lVjb@nPnSgPnOqVlSNNeA|LKtGa@jXgF?CjDYx^qG|JoJ)
- [x] [stroke, enc](http://localhost:8080/styles/osm-bright/static/auto/400x400.png?path=stroke:%233339|enc:ovvkC}ezyLEaC|DpRzDvRhBhLld@fO|]fUna@lVjb@nPnSgPnOqVlSNNeA|LKtGa@jXgF?CjDYx^qG|JoJ)
- [x] [enc](http://localhost:8080/styles/osm-bright/static/auto/400x400.png?path=enc:ovvkC%7DezyLEaC%7CDpRzDvRhBhLld@fO%7C%5DfUna@lVjb@nPnSgPnOqVlSNNeA%7CLKtGa@jXgF?CjDYx%5EqG%7CJoJ)

#### Coordinates
- [x] [fill, width, stroke, coordinates](http://localhost:8080/styles/osm-bright/static/auto/400x400.png?path=fill:%233f34|width:4|stroke:%233339|73.81468,18.55652%7C73.81469,18.55667%7C73.81469,18.55667%7C73.81388,18.55673%7C73.81388,18.55673)
- [x] [width, stroke, coordinates](http://localhost:8080/styles/osm-bright/static/auto/400x400.png?path=width:4|stroke:%233339|73.81468,18.55652%7C73.81469,18.55667%7C73.81469,18.55667%7C73.81388,18.55673%7C73.81388,18.55673)
- [x] [stroke, coordinates](http://localhost:8080/styles/osm-bright/static/auto/400x400.png?path=stroke:%233339|73.81468,18.55652%7C73.81469,18.55667%7C73.81469,18.55667%7C73.81388,18.55673%7C73.81388,18.55673)
- [x] [coordinates](http://localhost:8080/styles/osm-bright/static/auto/400x400.png?path=73.81468,18.55652%7C73.81469,18.55667%7C73.81469,18.55667%7C73.81388,18.55673%7C73.81388,18.55673)

#### Google
- [x] [enc](http://localhost:8080/styles/osm-bright/static/auto/400x400.png?path=width:4%7Cstroke:%233339%7Cenc:ovvkC%7DezyLEaC%7CDpRzDvRhBhLld@fO%7C%5DfUna@lVjb@nPnSgPnOqVlSNNeA%7CLKtGa@jXgF?CjDYx%5EqG%7CJoJ&path=fill:%233f34%7Cwidth:4%7Cstroke:%233339%7Cenc:ovvkC%7DezyLEaC%7CDpRzDvRhBhLld@fO%7C%5DfUna@lVjb@nPnSgPnOqVlSNNeA%7CLKtGa@jXgF?CjDYx%5EqG%7CJoJ)
#### Coordinates
- [x] [coordinates](http://localhost:8080/styles/osm-bright/static/auto/400x400.png?path=fill:%233f34|width:4|stroke:%233339|73.81468,18.55652%7C73.81469,18.55667%7C73.81469,18.55667%7C73.81388,18.55673%7C73.81388,18.55673&path=73.81468,18.55652%7C73.81469,18.55667%7C73.81469,18.55667%7C73.81388,18.55673%7C73.81388,18.55673)

Co-Authored-By: Niklas Hösl <nik.hoesl@hotmail.com>
Signed-off-by: Vinayak Kulkarni <19776877+vinayakkulkarni@users.noreply.github.com>

### CODE REVIEW

**Comment by acalcutt** [comment]
seems like a useful addition for long paths. I'll try to test it out soon. 

Anyone else have any opinions on this?

**Comment by vinayakkulkarni** [comment]
> seems like a useful addition for long paths. I'll try to test it out soon.
> 
> Anyone else have any opinions on this?

Instead of adding another arg `encodedpath`, can we make `path` to accept either lat,lng pairs or google encoded polyline?

**Comment by acalcutt** [comment]
Having them separate matches what maptiler is doing in their other products, like https://docs.maptiler.com/cloud/api/static-maps/ . I tend to want to stay in line with their other products, so it may make sense to keep it as it is.

**Comment by acalcutt** [comment]
Actually I notice they depreciated encodedpath and say just to use path. so maybe just having path is in line with what they are doing

**Comment by vinayakkulkarni** [comment]
> Actually I notice they depreciated encodedpath and say just to use path. so maybe just having path is in line with what they are doing

They've made it similar to Static Maps APIs which google provides. 

Will update the PR in some time

**Comment by acalcutt** [comment]
can you update the package-lock.json in this.

you can just do
npm i --ignore-scripts

**Comment by vinayakkulkarni** [comment]
> Did you still want to make path also support encoded path in this?

Yep, it’s in my local fs, will push tonight or maybe tomorrow after testing

File: src/serve_rendered.js [hunk 1]
```diff
@@ -147,47 +150,65 @@
...
+  const reqPath = decodeURIComponent(query.path);
+  const paths = [];
   // Return an empty list if no paths have been provided
-  if (!query.path) {
-    return [];
+  if ('path' in query && !query.path) {
+    return paths;
   }
+  // Parse paths provided via path query argument
+  if ('path' in query && PATH_PATTERN.test(reqPath)) {
```
[the remark this answers is not in the document]
Comment (vinayakkulkarni) [reply src/serve_rendered.js:160]: @acalcutt: I am planning on fixing those security issues which should fix this one as well.

**Comment by acalcutt** [comment]
Can you give a few example urls to test this? Does it affect of the examples in https://github.com/maptiler/tileserver-gl/pull/619 like https://github.com/maptiler/tileserver-gl/pull/619#issuecomment-1271756389

**Comment by acalcutt** [comment]
I tried to supply multiple paths in a URL and I got this crash

```
file:///opt/test9/enc_path/src/serve_rendered.js:162
      const splitPaths = query.path.split('|');
                                    ^

TypeError: query.path.split is not a function
    at extractPathsFromQuery (file:///opt/test9/enc_path/src/serve_rendered.js:162:37)
    at file:///opt/test9/enc_path/src/serve_rendered.js:1081:25
    at Layer.handle [as handle_request] (/opt/test9/enc_path/node_modules/express/lib/router/layer.js:95:5)
    at next (/opt/test9/enc_path/node_modules/express/lib/router/route.js:144:13)
    at Route.dispatch (/opt/test9/enc_path/node_modules/express/lib/router/route.js:114:3)
    at Layer.handle [as handle_request] (/opt/test9/enc_path/node_modules/express/lib/router/layer.js:95:5)
    at /opt/test9/enc_path/node_modules/express/lib/router/index.js:284:15
    at param (/opt/test9/enc_path/node_modules/express/lib/router/index.js:365:14)
    at param (/opt/test9/enc_path/node_modules/express/lib/router/index.js:376:14)
    at param (/opt/test9/enc_path/node_modules/express/lib/router/index.js:376:14)

```

**Comment by acalcutt** [comment]
For example http://192.168.0.103:8080/styles/basic-preview/static/auto/400x400.png?path=width:4%7Cstroke:%233339%7Cenc:ovvkC%7DezyLEaC%7CDpRzDvRhBhLld@fO%7C%5DfUna@lVjb@nPnSgPnOqVlSNNeA%7CLKtGa@jXgF?CjDYx%5EqG%7CJoJ&path=fill:%233f34%7Cwidth:4%7Cstroke:%233339%7Cenc:ovvkC%7DezyLEaC%7CDpRzDvRhBhLld@fO%7C%5DfUna@lVjb@nPnSgPnOqVlSNNeA%7CLKtGa@jXgF?CjDYx%5EqG%7CJoJ

**Review by acalcutt — APPROVED** [review]

**Comment by acalcutt** [comment]
Can you rebase this

**Comment by vinayakkulkarni** [comment]
> Can you rebase this / fix the package-lock.json conflict

This is done

**Pull request merged**